### ISSUE TITLE
reserved_ip_range expects one value on creation, but another on update

### ISSUE THREAD

**Issue by jamiezieziula** [terraform-google-modules/terraform-google-memorystore#89]
### TL;DR

On initial apply of a redis instance - the module expects the **name** of the reserved private ip space (ex. `ga-dev-shared-base-hub-vpc-peering-internal`).  On any susbsequent apply, it expects an actual cidr range (ex. `10.30.1.0/29`).

### Expected behavior

I would expect that providing the reserved IP space name to be consistent across initial apply and subsequent apply runs.

### Observed behavior

Terraform wants to recreate the instance when providing the IP space name, rather than the CIDR.

### Terraform Configuration

```hcl
module "redis" {
  source  = "terraform-google-modules/memorystore/google"
  version = "~> 4.3.0"

  name    = "redis"
  project = var.project_id

  tier          = "BASIC"

  region      = "us-east1"
  location_id = var.redis_zone

  authorized_network = data.terraform_remote_state.network.outputs.vpc_id
  reserved_ip_range       = "ga-dev-shared-base-hub-vpc-peering-internal"
  connect_mode            = "PRIVATE_SERVICE_ACCESS"
  transit_encryption_mode = "DISABLED"
}
```
```


### Terraform Version

```sh
1.2.3
```

### Additional information

<img width="1061" alt="Screen Shot 2022-06-25 at 1 13 12 PM" src="https://user-images.githubusercontent.com/21315568/175783902-859e0176-e5a7-473c-9550-d48dc3f3c486.png">

**Comment by fenos** [comment]
I have the same issue, is there a workaround?
Currently every time I apply the module it will need to re-create the instance

**Comment by jamiedick** [comment]
This is what I'm currently doing to handle this:
```
data "terraform_remote_state" "network" {
  backend = "remote"
  config = {
    organization = "orgname"
    workspaces = {
      name = "${var.env}-network"
    }
  }
}

data "google_redis_instance" "redis" {
  name   = "redis-${var.env}"
  region = var.region
}

module "redis" {
  source  = "terraform-google-modules/memorystore/google"
  version = "~> 4.3.0"

  reserved_ip_range       = data.google_redis_instance.redis.reserved_ip_range != null ? data.google_redis_instance.redis.reserved_ip_range : data.terraform_remote_state.network.outputs.private_service_address_name
}
```

**Comment by fenos** [comment]
Thanks @jamiedick 🙏
